src/functions/operatorname.js

```diff
@@ -3,115 +3,160 @@ import defineFunction, {ordargument} from "../defineFunction";
+    }
+};
+
+const mathmlBuilder: MathMLBuilder<"operatorname"> = (group, options) => {
+    // The steps taken here are similar to the html version.
+    let expression = mml.buildExpression(
+        group.body, options.withFont("mathrm"));
+
+    // Is expression a string or has it something like a fraction?
+    let isAllString = true;  // default
+    for (let i = 0; i < expression.length; i++) {
+        const node = expression[i];
+        if (node instanceof mathMLTree.SpaceNode) {
+            // Do nothing
+        } else if (node instanceof mathMLTree.MathNode) {
+            switch (node.type) {
+                case "mi":
+                case "mn":
+                case "ms":
+                case "mspace":
+                case "mtext":
+                    break;  // Do nothing yet.
+                case "mo": {
+                    const child = node.children[0];
+                    if (node.children.length === 1 &&
+                        child instanceof mathMLTree.TextNode) {
+                        child.text =
+                            child.text.replace(/\u2212/, "-")
+                                .replace(/\u2217/, "*");
+                    } else {
                         isAllString = false;
+                    }
+                    break;
                 }
-            } else {
-                isAllString = false;
+                default:
+                    isAllString = false;
             }
+        } else {
+            isAllString = false;
         }
+    }
 
-        if (isAllString) {
-            // Write a single TextNode instead of multiple nested tags.
-            const word = expression.map(node => node.toText()).join("");
-            expression = [new mathMLTree.TextNode(word)];
-        }
+    if (isAllString) {
+        // Write a single TextNode instead of multiple nested tags.
+        const word = expression.map(node => node.toText()).join("");
+        expression = [new mathMLTree.TextNode(word)];
+    }
 
-        const identifier = new mathMLTree.MathNode("mi", expression);
-        identifier.setAttribute("mathvariant", "normal");
+    const identifier = new mathMLTree.MathNode("mi", expression);
+    identifier.setAttribute("mathvariant", "normal");
 
-        // \u2061 is the same as &ApplyFunction;
-        // ref: https://www.w3schools.com/charsets/ref_html_entities_a.asp
-        const operator = new mathMLTree.MathNode("mo",
-            [mml.makeText("\u2061", "text")]);
+    // \u2061 is the same as &ApplyFunction;
+    // ref: https://www.w3schools.com/charsets/ref_html_entities_a.asp
+    const operator = new mathMLTree.MathNode("mo",
+        [mml.makeText("\u2061", "text")]);
 
+    if (group.parentIsSupSub) {
+        return new mathMLTree.MathNode("mo", [identifier, operator]);
+    } else {
         return mathMLTree.newDocumentFragment([identifier, operator]);
+    }
+};
+
+// \operatorname
+// amsopn.dtx: \mathop{#1\kern\z@\operator@font#3}\newmcodes@
```

Comment:
We don't actually follow this definition do we?

Reply:
Well, I've tried to emulate its behavior as closely as I can. I'll let others judge how well I did.

Reply:
Is `\kern\z@` a zero width kern to ensure a minimum height/depth?  I assume `\operator@font` is an alias for cmr.  I guess it useful to have the comment if we're not doing a literal translation of it.

Reply:
Re: `\kern\z`. According to `amsopn`:

> In the interior of the \mathop we need a null object (we choose a zero kern for
minimum waste of main mem) in order to guard against the case where #3 is a
single letter; TeX will seize it and center it on the math axis if there is nothing
else inside the \mathop atom.

Reply:
I didn't realize that TeX centered single letter on the math axis.  I just assumed that it used baseline alignment.  TIL.